src/main/java/io/opentelemetry/semconv/AttributeKeyTemplate.java

```diff
@@ -14,9 +14,9 @@
 
 public final class AttributeKeyTemplate<T> {
 
-  private final ConcurrentMap<String, AttributeKey<T>> keysCache = new ConcurrentHashMap<>();
   private final String prefix;
   private final Function<String, AttributeKey<T>> keyBuilder;
+  private ConcurrentMap<String, AttributeKey<T>> keysCache;
```

Comment:
I believe this would need to be `volatile` for thread safety. I'd suggest staying with `final` and inline initialization.

Reply:
Technically I think this is one of those weird situations where it doesn't actually matter if two threads read this as null and instantiate / use different instances of `ConcurrenHashMap`, since the consequence is just that you will have created two copies of the same `AttributeKey` instead of returning a cached copy.

The inline initialization is simpler though and doesn't raise flags about thread safety. I'd say stick with the `final` inline initialization and keep this type of thing in our back pocket if we get feedback about utilization.